mgmt/ProxyConfig.h

```diff
@@ -98,15 +98,31 @@ class ConfigProcessor
 
 // A Continuation wrapper that calls the static reconfigure() method of the given class.
 template <typename UpdateClass> struct ConfigUpdateContinuation : public Continuation {
+  void (*f)();
   int
   update(int /* etype */, void * /* data */)
   {
     UpdateClass::reconfigure();
     delete this;
     return EVENT_DONE;
   }
+  int
+  update_func(int /*etype*/, void * /* data */)
```

Comment:
Unfortunately if any thread is actively using the config structure the it will have its keyfile updated asynchronously. It seems likely to fail. At a minimum the key block pointer should be atomically swapped *after* the new key block is loaded.